packages/react/src/auto/polaris/inputs/PolarisAutoEnumInput.tsx

```diff
@@ -0,0 +1,149 @@
+import type { ComboboxProps } from "@shopify/polaris";
+import { AutoSelection, Combobox, EmptySearchResult, Icon, InlineStack, Listbox, Tag, Text } from "@shopify/polaris";
+import { SearchIcon } from "@shopify/polaris-icons";
+import React, { useEffect, useMemo } from "react";
+import { type Control } from "react-hook-form";
+import { useEnumInputController } from "../../hooks/useEnumInputController.js";
+
+export const PolarisAutoEnumInput = (props: { field: string; control?: Control<any> } & Partial<ComboboxProps>) => {
+  const { field: fieldApiIdentifier, control, ...comboboxProps } = props;
+  const {
+    allowMultiple,
+    allowOther,
+    onSelectionChange,
+    selectedOptions,
+    allOptions,
+    filteredOptions,
+    searchQuery,
+    label,
+    metadata,
+    isError,
+    errorMessage,
+  } = useEnumInputController({ field: fieldApiIdentifier, control });
+  const { value: searchValue, setValue: setSearchValue } = searchQuery;
+
+  const selectedTagsElement = useMemo(
+    () =>
+      allowMultiple && selectedOptions.length > 0 ? (
+        <InlineStack gap="150">
+          {selectedOptions.map((tag) => (
+            <Tag key={`option-${tag}`} onRemove={() => onSelectionChange(tag)}>
+              {tag}
+            </Tag>
+          ))}
+        </InlineStack>
+      ) : null,
+    [allowMultiple, onSelectionChange, selectedOptions]
+  );
+
+  const optionItemElement = useMemo(() => {
+    if (allOptions.length === 0) return null;
+
+    const formatOptionText = (option: string) => {
```

Comment:
there is probably a small performance optimization here to put this in a useCallback() separately from the useMemo